src/Components/Onboarding/Wizard.tsx

```diff
@@ -1,108 +1,74 @@
 import React from "react"
+import { Redirect, Route } from 'react-router'
+
 import Events from "../../Utils/Events"
 import { track } from "../../Utils/track"
 import { ProgressIndicator } from "./ProgressIndicator"
-import { StepComponent, StepSlugs } from "./Types"
 
-interface Props {
-  stepComponents: StepComponent[]
-  redirectTo?: string
-  forceStep?: StepSlugs
+import Artists from './Steps/Artists'
+import { BudgetComponent as Budget } from "./Steps/Budget"
+import { CollectorIntentComponent as CollectorIntent } from "./Steps/CollectorIntent"
+import Genes from './Steps/Genes'
+
+const STEPS = [
+  `/personalize/${CollectorIntent.slug}`,
+  `/personalize/${Artists.slug}`,
+  `/personalize/${Genes.slug}`,
+  `/personalize/${Budget.slug}`
+]
+
+export interface Props {
   tracking?: any
 }
 
-interface State {
-  currentStep: number
-  nextButtonEnabled: boolean
-  percentComplete: number
+export interface State {
+  finished: boolean
 }
 
 @track({}, { dispatch: data => Events.postEvent(data) })
-class Wizard extends React.Component<Props, State> {
-  static defaultProps = {
-    redirectTo: "/",
-    forceStep: null,
-  }
-
+export class Wizard extends React.Component<Props, State> {
   constructor(props) {
     super(props)
-
     this.state = {
-      currentStep: this.getForceStep(),
-      nextButtonEnabled: false,
-      percentComplete: 0,
+      finished: false,
     }
   }
 
-  getForceStep = () => {
-    const { forceStep } = this.props
-    const stepSlugs = this.props.stepComponents.map(step => step.slug)
-    if (forceStep && stepSlugs.includes(forceStep)) {
-      return stepSlugs.indexOf(forceStep)
-    } else {
-      return 0
-    }
-  }
-
-  componentDidMount() {
-    const { stepComponents } = this.props
-    const { currentStep } = this.state
-    window.history.pushState(
-      {},
-      "",
-      `/personalize/${stepComponents[currentStep].slug}`
-    )
-  }
-
-  getCurrentStep(): JSX.Element | null {
-    const currentStep = this.state.currentStep
-
-    if (currentStep > this.props.stepComponents.length - 1) {
-      return null
-    }
-
-    const CurrentStep = this.props.stepComponents[currentStep]
-    return <CurrentStep onNextButtonPressed={this.onNextButtonPressed} />
+  onNextButtonPressed = (increaseBy, history) => {
+    history.push(STEPS[STEPS.indexOf(location.pathname) + increaseBy])
   }
 
-  onFinish = () => {
-    const { redirectTo } = this.props
-    this.setState({ percentComplete: 1 })
-    setTimeout(() => {
-      window.location.href = redirectTo
-    }, 500)
+  onFinish = redirectTo => {
+    this.setState({ finished: true })
+    setTimeout(() => (window.location.href = redirectTo || "/"), 500)
 
     this.props.tracking.trackEvent({
       action: "Completed Onboarding",
     })
```

Comment:
I left the analytics code here so we should be good to go.

Reply:
Ohhhhh I see now. Using @track directly in Force was the problem. Not that the track calls in each step didn't work, just this one in the main wrapper. Yea that totes makes sense.